server/server.go

```diff
@@ -2418,8 +2411,26 @@ func (s *StanServer) processCloseRequest(m *nats.Msg) {
 		return
 	}
 
+	// In clustered mode, drop the request if we're not the metadata leader.
+	// Another server will handle it.
+	if s.isClustered() && !s.isMetadataLeader() {
+		return
+	}
+
+	// If clustered, thread operations through Raft.
+	if s.isClustered() {
+		if err := s.replicateConnClose(req.ClientID, m.Reply, true); err != nil {
```

Comment:
Here, server is metaleader and for it - only - the m.Reply matters. That is, this is the client's reply inbox the connect response/error needs to be sent to. This should not be used by followers.